modules/sitemaps/sitemaps.php

```diff
@@ -1,782 +1,1556 @@
+			serve_raw_and_die(
+				'text/xml',
+				Jetpack_Sitemap_Stylist::sitemap_index_xsl()
+			);
+		}
+
+		// Catch image sitemap xml
+		if ( preg_match( $regex['image'], $_SERVER['REQUEST_URI']) ) {
+			serve_raw_and_die(
+				'application/xml',
+				$this->get_contents_of_post(
+					substr($_SERVER['REQUEST_URI'], 1, -4),
+					'jp_img_sitemap'
+				)
+			);
+		}
+
+		// Catch image sitemap index xml
+		if ( preg_match( $regex['image-index'], $_SERVER['REQUEST_URI']) ) {
+			serve_raw_and_die(
+				'application/xml',
+				$this->get_contents_of_post(
+					substr($_SERVER['REQUEST_URI'], 1, -4),
+					'jp_img_sitemap_index'
+				)
+			);
+		}
+
+		// Catch image sitemap xsl
+		if ( preg_match( $regex['image-style'], $_SERVER['REQUEST_URI']) ) {
+			serve_raw_and_die(
+				'text/xml',
+				Jetpack_Sitemap_Stylist::image_sitemap_xsl()
+			);
+		}
+
+		// Catch news sitemap xml
+		if ( preg_match( $regex['news'], $_SERVER['REQUEST_URI']) ) {
+			serve_raw_and_die(
+				'text/xml',
+				$this->news_sitemap_xml()
+			);
+		}
+
+		// Catch news sitemap xsl
+		if ( preg_match( $regex['news-style'], $_SERVER['REQUEST_URI']) ) {
+			serve_raw_and_die(
+				'text/xml',
+				Jetpack_Sitemap_Stylist::news_sitemap_xsl()
+			);
+		}
+
+		// URL did not match any sitemap patterns.
+		return;
+	}
 
-/**
- * Define an array of attribute value pairs for use inside the root element of an XML document.
- * Intended for mapping namespace and namespace URI values.
- * Passes array through jetpack_sitemap_ns for other functions to add their own namespaces.
- *
- * @module sitemaps
- *
- * @return array array of attribute value pairs passed through the jetpack_sitemap_ns filter
- */
-function jetpack_sitemap_namespaces() {
 	/**
-	 * Filter the attribute value pairs used for namespace and namespace URI mappings.
+	 * Add actions to schedule sitemap generation.
+	 * Should only be called once, in the constructor.
 	 *
-	 * @module sitemaps
+	 * @access private
+	 * @since 4.5.0
+	 */
+	private function schedule_sitemap_generation () {
+		/**
+		 * Callback for adding sitemap-interval to the list of schedules.
+		 *
+		 * @since 4.5.0
+		 */
+		function callback_add_sitemap_schedule ($schedules) {
+			$schedules['sitemap-interval'] = array(
+				'interval' => Jetpack_Sitemap_Manager::SITEMAP_INTERVAL,
+				'display'  => __('Every Minute')
+			);
+			return $schedules;
+		}
+
+		// Add cron schedule
+		add_filter( 'cron_schedules', 'callback_add_sitemap_schedule');
+
+		add_action(
+			'jp_sitemap_cron_hook',
+			array( $this, 'build_all_sitemaps' )
+		);
+
+		if( !wp_next_scheduled( 'jp_sitemap_cron_hook' ) ) {
+			wp_schedule_event(
+				time(),
+				'sitemap-interval',
+				'jp_sitemap_cron_hook'
+			);
+		}
+
+		return;
+	}
+
+	/**
+	 * Callback to add sitemap to robots.txt.
 	 *
-	 * @since 3.9.0
+	 * @access public
+	 * @since 4.5.0
+	 */
+	public function callback_action_do_robotstxt () {
+		echo 'Sitemap: ' . home_url() . '/sitemap.xml' . PHP_EOL;
```

Comment:
You'll probably need to take the `jetpack_sitemap_location` filter into account, since some people may use the filter to define a custom sitemap location.